processing/src/main/java/org/apache/druid/frame/processor/ComposingOutputChannelFactory.java

```diff
@@ -114,7 +114,7 @@ public PartitionedOutputChannel openPartitionedChannel(String name, boolean dele
             }
           })::get;
       writableFrameChannelsBuilder.add(() -> channel.get().getWritableChannel());
-      readableFrameChannelSuppliersBuilder.add(() -> channel.get().getReadableChannelSupplier().get());
+      readableFrameChannelSuppliersBuilder.add(() -> channel.get().readOnly().getReadableChannelSupplier().get());
```

Comment:
This will open a readOnly channel for all ComposingOutputChannel's. That shouldn't be the case no ?

Reply:
Can we please add a comment saying that you should only request a readable channel if the outputchannel is completely written too.